src/main/java/pl/cinema/server/reservation/WeekendPriceEvaluator.java

```diff
@@ -0,0 +1,50 @@
+package pl.cinema.server.reservation;
+
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Component;
+
+import java.math.BigDecimal;
+import java.time.DayOfWeek;
+import java.time.LocalDateTime;
+import java.util.List;
+
+import static java.time.DayOfWeek.FRIDAY;
+import static java.time.DayOfWeek.SATURDAY;
+import static java.time.DayOfWeek.SUNDAY;
+import static java.time.LocalDateTime.now;
+
+@Component
+@RequiredArgsConstructor
+public class WeekendPriceEvaluator implements PriceEvaluator {
+    private static final List<DayOfWeek> weekendDays = List.of(FRIDAY, SATURDAY, SUNDAY);
+    private static final BigDecimal CURRENT_WEEKEND_CHARGE = BigDecimal.valueOf(4);
+    private final WeekendTicketsProperty ticketsProperty;
+
+    @Override
+    public Price evaluatePrice(String voucherCode) {
+        return new Price(CURRENT_WEEKEND_CHARGE);
```

Comment:
a gdzie jest discount brany pod uwagę?

Reply:
a obejmuje on także mniejszy dodatek w weekend? Myślałem, że weekendowa opłata jest stała / tak założyłem, ale jak chcemobniżyć całą cenę to trzeba by to zmienić oczywiście, dobra uwaga